### DISCUSSION TITLE
Discussion on recreating keploy.io navbar; Linked to - https://github.com/keploy/keploy/issues/2068

### DISCUSSION THREAD

**Discussion by yaten2302** [keploy/keploy#2079] [General]
https://github.com/keploy/keploy/issues/2068#issue-2408763585

Since, 4 fields in the navbar have to be created. what we can do is that 2 folks can work on the "Platform" part, since it's quite big and the other 2 can work on "About" and "Blogs" (since they are quite small and easy to handle). The "Docs" part which is left, ig anyone of us can implement that since only 1 link has to be added in it.

What do you think @PranavBarthwal @vaishnavi192 @Pratik772846?

**Comment by PranavBarthwal** [comment]
@yaten2302 yeah! sounds good. I am willing to work on About and docs section.

**Reply by yaten2302** [reply]
Great @PranavBarthwal 👍 
Also, what I think is that only one of us should create the final PR and others can edit on his/her fork? What do you think @PranavBarthwal @vaishnavi192 @Pratik772846 ? I've a fork created BTW, if you'll agree, then I can share it here and then we'll can make the changes in it and then create only 1 PR. Because, working on this issue separately ig would create problems and we won't be able to view each other's work?

**Reply by PranavBarthwal** [reply]
I guess, we should work separately on our forks. We can collaborate and decide which section will be built first, then only next person will start building other section.
What say?

**Reply by yaten2302** [reply]
@PranavBarthwal since we're working on a single NavBar, ig working on separate forks will be difficult since several features have to be grouped together and different folks working on different code might create problems. I hope you understand my concern? If this doesn't suits, then we can start working on the navbar separately 👍 
Others can also give their opinion so that we can start working on this issue tomorrow positively @PranavBarthwal ?
@vaishnavi192 @Pratik772846?

**Reply by PranavBarthwal** [reply]
Yeah, let's wait for other's replies.

**Reply by yaten2302** [reply]
@PranavBarthwal ig, should we start with the work today? By the time others reply, we'll complete our part. What do you think?
Also, can we connect on Slack for this?

**Reply by vaishnavi192** [reply]
@yaten2302 @PranavBarthwal Hey I think we should ask @Sonichigo  about it coz if we create different prs then it might result in conflicts also. But how will final contributions be counted in final dashboard? 2-2 people can work on separate section and raise pr separately after discussing with others in grp....that will be fine but single pr will be more convenient for them

**Reply by Sonichigo** [reply]
The Final Contribution in this case would be split equally among the contributors, and a single PR would be greatly appreciated since in case of conflicts it can be addressed easily.

**Reply by vaishnavi192** [reply]
Yeah right! Thanks. I can take up blogs part.

**Reply by yaten2302** [reply]
Great @vaishnavi192  👍 
Ok, so what we can do is that I've a repo already forked of the Keploy's website everyone can work on that and then I'll create the final PR and then ofc all the contributions which you all made will be visible in the PR. Here's the link - https://github.com/yaten2302/website
Or if any one of you wants to volunteer for making that 1 PR, feel free to do so, just send the link here, so that we can fork that and then work accordingly 👍 

@PranavBarthwal 

I'll take up the Platform part then.

**Reply by PranavBarthwal** [reply]
@yaten2302 we'll fork your repo only 👍

**Reply by yaten2302** [reply]
@PranavBarthwal @vaishnavi192 the navbar requires a [redesign](https://github.com/keploy/keploy/issues/2068#issuecomment-2233407277), I'll try to complete it by today and then we can start with the coding part👍

**Reply by PranavBarthwal** [reply]
Okaay

**Reply by yaten2302** [reply]
Also, @PranavBarthwal @vaishnavi192 if anyone of have a website or design similar to [this one](https://github.com/keploy/keploy/issues/2068#issuecomment-2233407277), could you please share it here, so that I can take help from that

**Reply by yaten2302** [reply]
Hey @harshsbhat , since Pratik isn't active, would you like to work on the Platform part with me? Pranav and Vaishnavi are doing for the Blogs and the About part. Since, platform is quite big, so we both can work on that?

**Reply by yaten2302** [reply]
@PranavBarthwal @vaishnavi192, the prototype is approved👍 we can start with the coding part now.

https://www.figma.com/design/n3BO0caZntCqDLe8CFCi3f/Keploy-NavBar?node-id=0-1&t=JFO7QnVRxbio7Uvw-1

**Reply by PranavBarthwal** [reply]
Cool 👍

**Reply by yaten2302** [reply]
@vaishnavi192 since you are doing for the `Blogs` part, there are 2 links in it:
1. Technology - https://keploy.io/blog/technology
2. Community - https://keploy.io/blog/community

**Reply by yaten2302** [reply]
@Swpn0neel @Sonichigo sir, could you please confirm that are the above links correct?
Also, for the `Platform` and `About` part, could you please guide that which links have to be added? Like, in the issue, the screenshot which you shared <br>
![image](https://github.com/user-attachments/assets/88bcec96-2fc3-406b-8fc7-9fe95dd06ae4)

In here, for both the sections, could you please guide that what all links should be added?

**Reply by yaten2302** [reply]
BTW @PranavBarthwal @vaishnavi192 , by the time we get the links to be put, we can start with the work now?

**Reply by PranavBarthwal** [reply]
Yess we should start

**Reply by yaten2302** [reply]
@vaishnavi192 , since you've the prototype ready with the links provided above, you can fork [this repo](https://github.com/yaten2302/website) and start you work, by the time we all will start without links as of now to complete this ASAP 👍

**Reply by Sonichigo** [reply]
> @Swpn0neel @Sonichigo sir, could you please confirm that are the above links correct? Also, for the `Platform` and `About` part, could you please guide that which links have to be added? Like, in the issue, the screenshot which you shared ![image](https://private-user-images.githubusercontent.com/129659514/350747522-88bcec96-2fc3-406b-8fc7-9fe95dd06ae4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2MzMwMjAsIm5iZiI6MTcyMTYzMjcyMCwicGF0aCI6Ii8xMjk2NTk1MTQvMzUwNzQ3NTIyLTg4YmNlYzk2LTJmYzMtNDA2Yi04ZmM3LTlmZTk1ZGQwNmFlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyMlQwNzE4NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNWZkNGZmOGQ3NTY1MjdkMTlkMTUwMjlhOWU0YzlmNTNmYTU2YmQwZmQ2ZDljYWNjZTMxY2Y2YjYzNjdjNjYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.vv3JPvXBd1k3DK2-vyputcjEqr92Vvpv4dtPS-jnyE0)
> 
> In here, for both the sections, could you please guide that what all links should be added?

1. USP - https://keploy.io/docs/keploy-explained/why-keploy/
2. What is Keploy -  https://keploy.io/docs/keploy-explained/introduction/
3. How Keploy Works - https://keploy.io/docs/keploy-explained/how-keploy-works/
4. Enterprise Edition - https://keploy.io/docs/keploy-cloud/cloud-installation/

Solution - 
1. API Testing. - https://keploy.io/api-testing
2. Regression Testing - https://keploy.io/regression-testing
3. Continuous Integration Testing - https://keploy.io/continuous-integration-testing
4. Unit Test Generation - https://keploy.io/unit-test-generator
5. AI Code Generation - https://keploy.io/ai-code-generation

**Reply by yaten2302** [reply]
@vaishnavi192 @PranavBarthwal , in [this repo](https://github.com/yaten2302/website/tree/main), I've changed the rendering of the heading - `components/nav/navItemWithSmallDropdown.tsx`. Like, earlier only the "Resources" was rendered in this file, but now what we can do is that we can use this as a common template for both the `Blogs` and `About` section, you can make changes according to the prototype given Figma and it'll be applied to both the sections 👍 
Since, the Platform part is quite big, so it requires an additional template to render. I'll take care of that 👍

**Reply by PranavBarthwal** [reply]
> @vaishnavi192 @PranavBarthwal , in [this repo](https://github.com/yaten2302/website/tree/main), I've changed the rendering of the heading - `components/nav/navItemWithSmallDropdown.tsx`. Like, earlier only the "Resources" was rendered in this file, but now what we can do is that we can use this as a common template for both the `Blogs` and `About` section, you can make changes according to the prototype given Figma and it'll be applied to both the sections 👍 Since, the Platform part is quite big, so it requires an additional template to render. I'll take care of that 👍

whats the status of navbar?

**Reply by yaten2302** [reply]
@PranavBarthwal , I'm working on the platform part and you and @vaishnavi192 will be working on the "About" and "Blogs" section ig?
Have you both started?

**Comment by harshsbhat** [comment]
Even I would like to work on this
